app/models/audit/event/authn.rb

```diff
@@ -0,0 +1,53 @@
+require 'ostruct'
+
+class Audit::Event::Authn < OpenStruct
+  def emit_success
```

Comment:
You don't need to fix this, but there's something confusing to me about the naming.  we have an "event" emitting a "success" (a "success event").  so an event is emitting an event, which reads oddly.  

There's also a kind of strange inversion in the design, in that this object delegates to the thing that actually writes the log.  Ie, we have an `Event` with an `Audit` (the actually log writer) inside of it, and the event calls its own `emit_success` which then calls Audit, passing the events own structured data to `Audit.info`.  

I think a simpler, clearer design would be to just have pass events to Audit's writer methods, and allow client code to use `Audit`.  But that's a larger (though not very large) refactor -- just food for thought.

Reply:
Spent some time thinking about it actually. Maybe it'll get refactored at some point.